src/lib/FlightTasks/tasks/Utility/StraightLine.cpp

```diff
@@ -0,0 +1,214 @@
+		// Vehicle has reached target. Lock position
+		position_setpoint = _target;
+		velocity_setpoint = Vector3f(0.0f, 0.0f, 0.0f);
+		_is_target_reached = true;
+
+		return;
+	}
+
+	// unit vector in the direction of the straight line
+	Vector3f u_orig_to_target = (_target - _origin).unit_or_zero();
+	// vector from origin to current position
+	Vector3f orig_to_pos = _pos - _origin;
+	// current position projected perpendicularly onto desired line
+	Vector3f closest_pt_on_line = _origin + u_orig_to_target * (orig_to_pos * u_orig_to_target);
+	// previous velocity in the direction of the line
+	float speed_sp_prev = math::max(velocity_setpoint * u_orig_to_target, 0.0f);
+
+	// Calculate braking distance depending on speed, speed at target and deceleration (add 10% safety margin)
+	float braking_distance = 1.1f * (powf(_desired_speed - _desired_speed_at_target, 2) / (2.0f * _desired_deceleration));
+
+	float dist_to_target = (_target - _pos).length(); // distance to target
+
+	// Either accelerate or decelerate
+	float speed_sp    = dist_to_target > braking_distance ? _desired_speed        :  _desired_speed_at_target;
+	float max_acc_dec = dist_to_target > braking_distance ? _desired_acceleration : -_desired_deceleration;
+
+	float acc_track = (speed_sp - speed_sp_prev) / _deltatime;
+
+	if (fabs(acc_track) > fabs(max_acc_dec)) {
+		// accelerate/decelerate with desired acceleration/deceleration towards target
+		speed_sp = speed_sp_prev + max_acc_dec * _deltatime;
+	}
+
+	// constrain the velocity
+	speed_sp = math::constrain(speed_sp, 0.0f, _desired_speed);
+
+	// set the position and velocity setpoints
+	position_setpoint = closest_pt_on_line;
+	velocity_setpoint = u_orig_to_target * speed_sp;
+
+}
+
+float StraightLine::getMaxAcc()
+{
+	// unit vector in the direction of the straight line
+	Vector3f u_orig_to_target = (_target - _origin).unit_or_zero();
+
+	// calculate the maximal horizontal acceleration
+	float divider = (sqrt(powf(u_orig_to_target(0), 2) + powf(u_orig_to_target(1), 2)));
+	float max_acc_hor = MPC_ACC_HOR_MAX.get();
+
+	if (divider > FLT_EPSILON) {
+		max_acc_hor /= divider;
+
+	} else {
+		max_acc_hor *= 1000.0f;
+	}
+
+	// calculate the maximal vertical acceleration
+	float max_acc_vert_original = u_orig_to_target(2) < 0 ? MPC_ACC_UP_MAX.get() : MPC_ACC_DOWN_MAX.get();
+	float max_acc_vert = max_acc_vert_original;
+
+	if (fabs(u_orig_to_target(2)) > FLT_EPSILON) {
+		max_acc_vert /= fabs(u_orig_to_target(2));
+
+	} else {
+		max_acc_vert *= 1000.0f;
+	}
+
+	return math::min(max_acc_hor, max_acc_vert);
+}
+
+float StraightLine::getMaxVel()
+{
+	// unit vector in the direction of the straight line
+	Vector3f u_orig_to_target = (_target - _origin).unit_or_zero();
+
+	// calculate the maximal horizontal velocity
+	float divider = (sqrt(powf(u_orig_to_target(0), 2) + powf(u_orig_to_target(1), 2)));
+	float max_vel_hor = MPC_XY_VEL_MAX.get();
+
+	if (divider > FLT_EPSILON) {
+		max_vel_hor /= divider;
```

Comment:
I am bit confused about that.
max_vel_hor is lower the larger the distance is from origin to target?